app/models/agent.rb

```diff
@@ -386,7 +398,11 @@ def receive!(options={})
 
         Agent.where(:id => agents_to_events.keys).each do |agent|
           agent.update_attribute :last_checked_event_id, event_ids.max
-          Agent.async_receive(agent.id, agents_to_events[agent.id].uniq)
+          if agent.no_bulk_receive?
+            agents_to_events[agent.id].uniq.each { |event_id| Agent.async_receive(agent.id, [event_id]) }
+          else
+            Agent.async_receive(agent.id, agents_to_events[agent.id].uniq)
```

Comment:
I was thinking about it, but the additional allocation made me not do it at first. Traded it by removing a different one :smile: